ts/util/table-builder.ts

```diff
@@ -255,6 +256,11 @@ export class TableBuilder<ParentType, ChildType> {
                 reload();
               })
           );
+          if (!filterSuboptions) {
+            throw new Error(
+              `Dropdown filter ${filter.key} has no menu options`
+            );
+          }
```

Comment:
Move this check above the `const filterSuboptions...` line and change to `if (!filter.values || !filter.values.length)`. Then you don't need the `?` in `filter.values.map`, and it'll also protect against `filter.values` being an empty array